implementations/rust/ockam/ockam_command/src/tcp/outlet/delete.rs

```diff
@@ -43,37 +63,94 @@ pub async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, DeleteCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_opts.at_node);
-    let node_name = parse_node_name(&node_name)?;
-    let node = BackgroundNode::create(&ctx, &opts.state, &node_name).await?;
-
-    // Check if there an outlet with the provided alias/name exists
-    let alias = cmd.alias;
-    node.ask_and_get_reply::<_, OutletStatus>(&ctx, Request::get(format!("/node/outlet/{alias}")))
-        .await?
-        .found()
-        .into_diagnostic()?
-        .ok_or(miette!(
-            "TCP outlet with alias {alias} was not found on Node {node_name}"
-        ))?;
-
-    // Proceed with the deletion
-    if opts.terminal.confirmed_with_flag_or_prompt(
-        cmd.yes,
-        "Are you sure you want to delete this TCP outlet?",
-    )? {
-        node.tell(&ctx, Request::delete(format!("/node/outlet/{alias}")))
-            .await?;
-
-        opts.terminal
+    DeleteTui::run(ctx, opts, cmd).await
+}
+
+#[ockam_core::async_trait]
+impl DeleteCommandTui for DeleteTui {
+    const ITEM_NAME: &'static str = "tcp-outlet";
+
+    fn cmd_arg_item_name(&self) -> Option<&str> {
+        self.cmd.alias.as_deref()
+    }
+
+    fn cmd_arg_delete_all(&self) -> bool {
+        self.cmd.all
+    }
+
+    fn cmd_arg_confirm_deletion(&self) -> bool {
+        self.cmd.yes
+    }
+
+    fn terminal(&self) -> Terminal<TerminalStream<Term>> {
+        self.opts.terminal.clone()
+    }
+
+    async fn get_arg_item_name_or_default(&self) -> miette::Result<String> {
+        let node_name = get_node_name(&self.opts.state, &self.cmd.node_opts.at_node);
+        let node_name = parse_node_name(&node_name)?;
+        self.cmd.alias.clone().ok_or(miette::miette!(
+            "Missing the tcp-outlet {} on node {}",
+            "ALIAS".light_magenta(),
+            node_name.light_magenta()
+        ))
```

Comment:
We don't have a concept of a default alias for TCP outlets, so in this case, you should either return the value from `self.cmd_arg_item_name()` or return an error. The same happens in the [`relay delete` command](https://github.com/build-trust/ockam/blob/54bb211c0f1eb20af324cc5e37b38f3a43ff9817/implementations/rust/ockam/ockam_command/src/relay/delete.rs#L97).

Reply:
@adrianbenavides sorry I don't understand this. 
Shouldn't be that when the terminal is not interactive and no `ALIAS` argument is provided a usage error is reported to the user saying that an alias should be provided on the default node (if `--at` flag isn't specified)?
![default_node](https://github.com/build-trust/ockam/assets/90763809/11accf6c-c6f7-4304-a298-adc26550a520)
![node_b](https://github.com/build-trust/ockam/assets/90763809/82bba614-cd5c-4d45-8df9-f7c560ab48e8)

Reply:
Note that the subject of this command is a TCP outlet and not a node, so the "default" item is referencing a TCP outlet, in this case, its alias.

We currently persist the TCP outlets for a node, but not which one is the default. So, in this command, we want to return an error in this function if the alias is not provided. This should only affect non-interactive usages.

Also, the fact that the TCP outlet doesn't exist in a node because the `alias` argument is None is false.